source/http_filesystem/http_filesystem_test.go

```diff
@@ -0,0 +1,183 @@
+package httpfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	st "github.com/golang-migrate/migrate/v4/source/testing"
+)
+
+func Test(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "")
```

Comment:
Use [bytes.Buffer](https://golang.org/pkg/bytes/#Buffer) instead of a temp dir since httpfs doesn't necessarily work with local files.